megatron/testing_utils.py

```diff
@@ -212,6 +213,23 @@ def torch_assert_equal(actual, expected):
     torch.testing.assert_close(actual, expected, rtol=0.0, atol=0.0)
 
 
+def is_torch_bf16_available():
+    # from https://github.com/huggingface/transformers/blob/26eb566e43148c80d0ea098c76c3d128c0281c16/src/transformers/file_utils.py#L301
+    if is_torch_available():
+        import torch
+        if not torch.cuda.is_available() or torch.version.cuda is None:
+            return False
+        if torch.cuda.get_device_properties(torch.cuda.current_device()).major < 8:
+            return False
+        if int(torch.version.cuda.split(".")[0]) < 11:
+            return False
+        if not version.parse(torch.__version__) > version.parse("1.09"):
```

Comment:
```suggestion
        if not version.parse(torch.__version__) >= version.parse("1.09"):
```

pt-1.9 already provides bf16 support, pt-1.10 for amp/bf16

will fix upstream.